tensorflow/src/main/scala/magnolify/tensorflow/ExampleType.scala

```diff
@@ -39,15 +45,22 @@ object ExampleType {
   def apply[T](cm: CaseMapper)(implicit f: ExampleField.Record[T]): ExampleType[T] =
     new ExampleType[T] {
       private val caseMapper: CaseMapper = cm
+      @transient override lazy val schema: Schema = f.schema(null)(caseMapper)
       override def from(v: Example): T = f.get(v.getFeatures, null)(caseMapper).get
       override def to(v: T): Example.Builder =
         Example.newBuilder().setFeatures(f.put(Features.newBuilder(), null, v)(caseMapper))
     }
 }
 
 sealed trait ExampleField[T] extends Serializable {
+  @transient private lazy val schemaCache: concurrent.Map[ju.UUID, Schema] =
+    concurrent.TrieMap.empty
   def get(f: Features, k: String)(cm: CaseMapper): Value[T]
   def put(f: Features.Builder, k: String, v: T)(cm: CaseMapper): Features.Builder
+
+  def schema(k: String)(cm: CaseMapper): Schema =
+    schemaCache.getOrElseUpdate(cm.uuid, schemaInternal(k)(cm))
+  protected def schemaInternal(k: String)(cm: CaseMapper): Schema
```

Comment:
maybe we can rename this "createSchema" or "buildSchema" so it's clearer what the function is?

Reply:
makes sense, will change